src/main/java/org/apache/brooklyn/cloudfoundry/entity/VanillaPaasApplicationDriver.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.entity;
+
+import java.util.Map;
+
+import org.apache.brooklyn.api.entity.drivers.EntityDriver;
+
+public interface VanillaPaasApplicationDriver extends EntityDriver {
+
+    boolean isRunning();
+
+    void rebind();
+
+    void start();
+
+    void restart();
+
+    void stop();
+
+    void delete();
+
+    void setEnv(Map<String, String> env);
```

Comment:
isn't that an async operation? do you think it makes sense to consider something like
```
ProcessTaskWrapper<Integer> setEnv(Map<String, String> env);
```